scheduler/reconcile.go

```diff
@@ -408,33 +451,36 @@ func (a *allocReconciler) computeGroup(group string, all allocSet) bool {
 
 	// Stop any unneeded allocations and update the untainted set to not
 	// include stopped allocations.
-	canaryState := dstate != nil && dstate.DesiredCanaries != 0 && !dstate.Promoted
-	stop := a.computeStop(tg, nameIndex, untainted, migrate, lost, canaries, canaryState, lostLaterEvals)
+	isCanarying := deploymentState != nil && deploymentState.DesiredCanaries != 0 && !deploymentState.Promoted
```

Comment:
#Refactoring review

Reply:
`isCanarying` 👍 

It's weirdly ungrammatical but I think it actually describes the state nicely.